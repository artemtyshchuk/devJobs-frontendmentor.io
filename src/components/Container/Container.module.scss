.headerImage {
  width: 100%;
  height: 160px;
  background-color: #5964e0;
  border-radius: 0 0 0 100px;
  position: absolute;
  z-index: -1;
}

.container {
  width: 80%;
  background: #fff;
  border-radius: 16px;
  margin: 0 auto;
}
