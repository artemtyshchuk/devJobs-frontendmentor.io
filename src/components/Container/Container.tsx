import { ReactNode } from "react";

import styles from "./Container.module.scss";

interface ContainerProps {
  children: ReactNode;
}

export const Container = ({ children }: ContainerProps) => {
  return (
    <div>
      <div className={styles.headerImage} data-testid="headerImage"></div>
      <div className={styles.container}>{children}</div>
    </div>
  );
};
