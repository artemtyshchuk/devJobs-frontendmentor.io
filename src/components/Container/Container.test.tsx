import { render, screen } from "@testing-library/react";
import { Container } from "./Container";

describe("Container", () => {
  it("should render Container with children", () => {
    render(<Container children={<p>Hello Container</p>} />);

    expect(screen.getByText("Hello Container")).toBeInTheDocument();
    expect(screen.getByTestId("headerImage")).toMatchSnapshot();
  });
});
