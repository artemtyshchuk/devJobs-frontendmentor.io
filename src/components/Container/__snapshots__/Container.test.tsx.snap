// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should render Container with children 1`] = `
<div
  class="headerImage"
  data-testid="headerImage"
/>
`;
